runtime/criusupport/criusupport.cpp

```diff
@@ -347,8 +351,143 @@ releaseSafeOrExcusiveVMAccess(J9VMThread *currentThread, J9InternalVMFunctions *
 	}
 }
 
+static char*
+endLine(char *cursor)
+{
+	while(true) {
+		switch (*cursor) {
+		case '\r':
+		case '\n':
+		case '\0':
+			if ((cursor[0] == '\r' && cursor[1] == '\n')
+			|| (cursor[0] == '\n')
+			) {
+				cursor[0] = '\0';
+			}
+			goto done;
+		default:
+			break;
+		}
+		cursor++;
+	}
+done:
+	return cursor;
+}
+
+static UDATA
+parseEnvVarsForOptions(J9VMThread *currentThread, J9JavaVMArgInfoList *vmArgumentsList, char *envFileChars, I_64 fileLength)
```

Comment:
I'm not sure this fits in this PR given that the description says:
> Documentation will be added later to describe what kind of options can
be specified upon restore. Initially, only java properties are fully
supported.

Wouldn't an initial implementation only do the `-D` option parsing (the java part of this PR)?

Reply:
The intention is to parse all options but only act on properties in this PR.

Reply:
Ill update the description

Reply:
> Documentation will be added later to describe what kind of options can
be specified upon restore. Initially all options will be parsed, but this PR 
only supports the addition of Java properties, all other options will
be ignored.

Reply:
I've modified it such that any specified on restore which is not consumed by the JVM will throw a restore exception. The only exception is if `-XX:+IgnoreUnrecognizedRestoreOptions` is specified as one of the restore options, in this case the VM will not complain about unconsumed options.